### PR TITLE
Validate harmonic angle idxs, refactor idx canonicalization

### PR DESCRIPTION
**Pull request by maxentile**
Based on suggestion https://github.com/proteneer/timemachine/pull/742#issuecomment-1128356628 , adds a check for completeness of the `angle_idxs` returned by `HarmonicAngleHandler`. This raises an error unless angle terms are present for all canonical angle tuples `(a, b, c)`.

Side quest: Also attempted to improve naming of index canonicalization function (`canonicalize_bond((a, b, c))` -> `canonicalize_angle((a, b, c))` or `canonicalize_bonded_ixn((a, b, c))`).

Side-side quest: Ran into a hiccup while adding `canonicalize_improper_torsion` -- will need to look more carefully at the logic of the two sites I'm extracting the definition from, or skip adding this for now.

### CODE REVIEW

File: timemachine/ff/handlers/utils.py [hunk 1]
```diff
@@ -32,6 +39,68 @@ def canonicalize_bond(arr):
...
+    a, c, d = sorted([_a, _c, _d])
+    canonicalized_1 = container_type([a, b, c, d])
+
+    # approach 2: take min((b, a, c, d), (d, c, a, b)) (used for applying trefoil convention)
+    a, b, c, d = torsion
+    canonicalized_2 = canonicalize_bonded_ixn([b, a, c, d])
+
+    # TODO: which one?
+    # return canonicalized
+    raise NotImplementedError(f"can't decide between {canonicalized_1} and {canonicalized_2}!")
```
Comment (maxentile) [inline timemachine/ff/handlers/utils.py:81]: I think it makes sense to go with approach 1 here, and possibly add a separate function to enumerate the trefoil permutations

Comment (maxentile) [reply timemachine/ff/handlers/utils.py]: Have now split this into a couple functions, and added documentation on points that tripped me up

**Comment by maxentile** [comment]
Superseded by #762

**Comment by proteneer** [comment]
Note that #762 operates at the level of the "combined single topology" molecule, not on the handler level.

**Pull request closed without merging**